pkg/kubecost/allocation.go

```diff
@@ -1357,14 +1373,9 @@ func (as *AllocationSet) AggregateBy(aggregateBy []string, options *AllocationAg
 			log.DedupedWarningf(3, "AllocationSet.AggregateBy: missing idleId for allocation: %s", alloc.Name)
 		}
 
-		skip := false
-
 		// (3) If the allocation does not match the filter, immediately skip the
 		// allocation.
-		if options.Filter != nil {
-			skip = !options.Filter.Matches(alloc)
-		}
-		if skip {
+		if !filter.Matches(alloc) {
```

Comment:
That's nice. `Matches` returns `true` if `filter` is `nil`? Or we just know it won't be `nil` here?

Reply:
Nevermind, I see where we make sure it's not nil.